src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -107,11 +107,6 @@ internal override JsonParameterInfoValues[] GetParameterInfoValues()
             JsonParameterInfoValues[] array;
             if (CtorParamInitFunc == null || (array = CtorParamInitFunc()) == null)
             {
-                if (SerializeHandler == null)
```

Comment:
Throwing an exception at configuration time is no longer needed -- this is generally done on serialization by checking the `MetadataSerializationNotSupported` property (set in line 116).